testing/repository_client.go

```diff
@@ -0,0 +1,93 @@
+package vtesting
+
+import (
+	"context"
+    "regexp"
+	"github.com/Shopify/voucher/repository"
+)
+const (
+    githubRegex = "^(?:https?|git)(?:://|@)(?P<vcs>[^/:]+)[/:](?P<org>[^/.]+)/(?P<name>[^/.]+)(?:.git)?"
+    commitRegex = "^(?:https?|git)(?:://|@)(?P<vcs>[^/:]+)[/:](?P<org>[^/.]+)/(?P<name>[^/.]+)/(?:commit)/(?P<hash>[^/.]+)"
+)
+
+type repositoryclient struct {
+    repos map[string]Repository
+}
+
+//NewClient creates a new repository client
+func NewClient() (RepoClient, error) {
+	return &repositoryclient{}, nil
+}
+
+//AddRepository adds to repository to the repository client
+func (c *repositoryclient) AddRepository(org repository.Organization, name string, commits map[string]repository.Commit, branches []repository.Branch) {
+    if(c.repos == nil) {
+        c.repos = make(map[string]Repository)
+    }
+    c.repos[org.Name] = Repository{org, name, commits, branches}
+}
+
+//GetCommitInfo creates commit info with provided
+func (c *repositoryclient) GetCommit(ctx context.Context, commitURL string) (repository.Commit, error) {
+    var commit repository.Commit
+    if len(c.repos) == 0 {
+    	if commitURL == "" {
+    		commit = repository.Commit{"https://github.com/org1/project/commit/abcd", nil, "good", false, nil}
+    	} else {
+    		commit = repository.Commit{commitURL, nil, "bad", true, nil}
+    	}
+    } else {
+        re, _ := regexp.Compile(commitRegex)
+        res := re.FindStringSubmatch(commitURL)
+        org_name := res[2]
+        repo := c.repos[org_name]
+        commit = repo.commits[commitURL]
+    }
+	return commit, nil
+}
+
+//GetOrganization creates organization
+func (c *repositoryclient) GetOrganization(ctx context.Context, uri string) (repository.Organization, error) {
+    var org repository.Organization
+	if uri == "" {
+		org = repository.CreateNewOrganization("1","test_org1", "https://test_org1.com")
+	} else {
+        re, _ := regexp.Compile(githubRegex)
+        org_name := re.FindStringSubmatch(uri)[2]
+        for i:=0; i < len(c.repos); i++ {
+            if _, repo := c.repos[org_name]; repo{
+                org = c.repos[org_name].org
+                break
+            }
+        }
+	}
+	return org, nil
+}
+
+func (c *repositoryclient) GetBranch(ctx context.Context, commitURL string) (repository.Branch, error) {
+	var branch repository.Branch
+    if len(c.repos) == 0 {
+    	if commitURL == "" {
+    		branch = repository.CreateNewBranch("test_def_branch1", nil)
+    	} else {
+    		branch = repository.CreateNewBranch("test_def_branch2", []repository.CommitRef{repository.CommitRef{commitURL+"1"}, 
+    			repository.CommitRef{commitURL+"2"}, repository.CommitRef{commitURL+"3"}})
+    	}
+    } else {
+        re, _ := regexp.Compile(githubRegex)
+        org_name := re.FindStringSubmatch(commitURL)[2]
+        
+        branches := c.repos[org_name].branches
+        for i:=0; i < len(branches); i++ {
```

Comment:
You can simplify this by running:

```suggestion
        for branchName, branch := range  c.repos[org_name].branches {
```

You can then interact with `branch.CommitRefs` directly!

Reply:
Done